### PR TITLE
[fix](runtime filter) Fix runtime filter not found

### PR DESCRIPTION
**Pull request by Gabriel39**
### What problem does this PR solve?

Problem Summary:

A runtime filter may have multiple targets and some of those are local-merge RF and others are not.
So for all runtime filters' producers, `publish` should notify all consumers in global RF mgr which manages local-merge RF and local RF mgr which manages others.

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [x] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [x] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [x] Confirm the release note
- [x] Confirm test cases
- [x] Confirm document
- [x] Add branch pick label

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Review by BiteTheDDDDt — DISMISSED** [review]

File: be/src/exprs/runtime_filter.cpp [hunk 1]
```diff
@@ -996,14 +996,13 @@ Status IRuntimeFilter::publish(bool publish_local) {
...
+        RETURN_IF_ERROR(_state->global_runtime_filter_mgr()->get_merge_addr(&addr));
         return filter->push_to_remote(&addr);
     };
-    auto send_to_local = [&](std::shared_ptr<RuntimePredicateWrapper> wrapper) {
-        std::vector<std::shared_ptr<IRuntimeFilter>> filters;
-        RETURN_IF_ERROR(_state->runtime_filter_mgr->get_consume_filters(_filter_id, filters));
-        DCHECK(!filters.empty());
-        // push down
+    auto send_to_local = [&](std::shared_ptr<RuntimePredicateWrapper> wrapper, bool global) {
+        std::vector<std::shared_ptr<IRuntimeFilter>> filters =
```
Comment (HappenLee) [inline be/src/exprs/runtime_filter.cpp:1003]: why not add new api `get_consume_filters` ？

File: be/src/exprs/runtime_filter.cpp [hunk 2]
```diff
@@ -996,14 +996,13 @@ Status IRuntimeFilter::publish(bool publish_local) {
...
-        RETURN_IF_ERROR(_state->runtime_filter_mgr->get_merge_addr(&addr));
+        RETURN_IF_ERROR(_state->global_runtime_filter_mgr()->get_merge_addr(&addr));
         return filter->push_to_remote(&addr);
     };
-    auto send_to_local = [&](std::shared_ptr<RuntimePredicateWrapper> wrapper) {
-        std::vector<std::shared_ptr<IRuntimeFilter>> filters;
-        RETURN_IF_ERROR(_state->runtime_filter_mgr->get_consume_filters(_filter_id, filters));
-        DCHECK(!filters.empty());
-        // push down
+    auto send_to_local = [&](std::shared_ptr<RuntimePredicateWrapper> wrapper, bool global) {
```
Comment (HappenLee) [inline be/src/exprs/runtime_filter.cpp:1002]: so here need use global ？ why still name `send_to_local` ?

File: be/src/exprs/runtime_filter.cpp [hunk 3]
```diff
@@ -1095,13 +1097,16 @@ class SyncSizeClosure : public AutoReleaseClosure<PSendFilterSizeRequest,
...
-    if (_need_local_merge) {
+    if (!_state->global_runtime_filter_mgr()->get_consume_filters(_filter_id).empty()) {
         LocalMergeFilters* local_merge_filters = nullptr;
-        RETURN_IF_ERROR(_state->runtime_filter_mgr->get_local_merge_producer_filters(
+        RETURN_IF_ERROR(_state->global_runtime_filter_mgr()->get_local_merge_producer_filters(
                 _filter_id, &local_merge_filters));
         std::lock_guard l(*local_merge_filters->lock);
         local_merge_filters->merge_size_times--;
         local_merge_filters->local_merged_size += local_filter_size;
+        if (_has_local_target) {
```
Comment (HappenLee) [inline be/src/exprs/runtime_filter.cpp:1107]: the code is wired, why here use global rf mgr get filter? has_local_target?

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Review by HappenLee — DISMISSED** [review]
LGTM

**Review by BiteTheDDDDt — DISMISSED** [review]

**Comment by Gabriel39** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by BiteTheDDDDt — APPROVED** [review]

**Pull request merged**